src/main/java/Duke.java

```diff
@@ -2,93 +2,166 @@
+            if(commandLength == VALID_TWO_PART_COMMAND && brokenDownInput[1].isEmpty()) {
+                brokenDownInput[0] = "invalid";
+            }
 
-        Task[] taskList = new Task[100];
-        int taskIndex = 0;
+            if(commandLength == VALID_TWO_PART_COMMAND && brokenDownInput[0].equals("done")) {
+                if(Integer.parseInt(brokenDownInput[1]) > 0 && Integer.parseInt(brokenDownInput[1]) <= taskCount) {
+                    markCompletion(tasks, brokenDownInput[1]);
+                    isValid = true;
+                }
+            }
 
-        String command;
-        command = in.nextLine();
-        boolean isFinish = false;
+            if(commandLength == VALID_ONE_PART_COMMAND && brokenDownInput[0].equals("list")) {
+                listTask(tasks, taskCount);
+                isValid = true;
+            }
 
-        //while the command is not bye it keeps asking for more
-        while(!command.equals("bye")) {
-            String[] breakDown = command.split(" ");
-            int size = breakDown.length;
-            //If its a 2 part command and first is done second is digit
-            if(size == 2 && breakDown[0].equals("done")) {
-                int taskNumber = Integer.parseInt(breakDown[1]);
-                int arrayIndex = taskNumber-1;
-                taskList[arrayIndex].markDone();
-                printLine();
-                System.out.println("Oooh I see you've done task " + taskNumber);
-                printLine();
-                isFinish = true;
+            if(commandLength == VALID_TWO_PART_COMMAND && brokenDownInput[0].equals("todo")) {
+                tasks[taskCount] = new Todo(brokenDownInput[1]);
+                taskCount = addTask(taskCount, ("[T] " + brokenDownInput[1]));
+                isValid = true;
             }
 
-            if(command.equals("list")) {
-                if(taskIndex > 0) {
-                    listTask(Arrays.copyOf(taskList,taskIndex));
+            if(commandLength == VALID_TWO_PART_COMMAND && brokenDownInput[0].equals("deadline")) {
+                String[] brokenDownBy = brokenDownInput[1].split(" /by ", 2);
+                if(brokenDownInput[1].contains(" /by ") && !brokenDownBy[1].isEmpty()) {
+                    tasks[taskCount] = new Deadline(brokenDownBy[0], brokenDownBy[1]);
+                    taskCount = addTask(taskCount, ("[D] " + brokenDownBy[0] + "(by: " + brokenDownBy[1]) + ")");
+                    isValid = true;
+                }
+            }
+            if(commandLength == VALID_TWO_PART_COMMAND && brokenDownInput[0].equals("event")) {
+                String[] brokenDownAt = brokenDownInput[1].split(" /at ", 2);
+                if(brokenDownInput[1].contains(" /at ") && !brokenDownAt[1].isEmpty()) {
+                    tasks[taskCount] = new Event(brokenDownAt[0], brokenDownAt[1]);
+                    taskCount = addTask(taskCount, ("[E] " + brokenDownAt[0] + "(at: " + brokenDownAt[1]) + ")");
```

Comment:
This one is a bit complicated. Can consider create a parser function just for this task